R/loadComp.R

```diff
@@ -244,11 +244,20 @@ predictSolute.loadComp <- function(
   # linearInterpolation and others that are supposed to run straight through
   # each residual.
   predvec_lin <- if(load.model@fit@log.resids) exp(preds) else preds
+  predvec_log <- if(load.model@fit@log.resids) preds else log(preds)
+  
+  # return preds in log or linear space
+  predvec <- if(lin.or.log == "log") predvec_log else predvec_lin
   
   # Change flux/conc formats if appropriate. MSEs are already formatted.
-  predvec_lin <- formatPreds(
-    predvec_lin, from.format=load.model@pred.format, to.format=flux.or.conc, 
-    newdata=newdata, metadata=load.model@metadata, attach.units=attach.units)
+  predvec <- formatPreds(
+    predvec, from.format=load.model@pred.format, to.format=flux.or.conc, 
+    newdata=newdata, metadata=load.model@metadata, lin.or.log=lin.or.log,
+    attach.units=attach.units)
+  
+  #\\# Do we need to force uncertainty calc if going from
+  #\\#      linear model to log return * OR *
+  #\\#      log model to linear return
```

Comment:
oh, gotchya. different things. 

we should calculate uncertainty whenever it's requested by the arguments `interval`, `se.fit`, or `se.pred`. uncertainty describes the range of values we think are reasonably probable. this range includes the main predictions but expands below and above them. this range is different in log and linear space, though the two can be linked. 

that link is where bias correction comes in, because both the prediction and the uncertainty are needed in linear space to come up with a prediction or uncertainty in log space, and vice versa. that link can be made by relatively simple math (as in `logToLin`) or by fancier/alternative methods - e.g., the Duan smearing estimator, which isn't in loadflex for now, or the MVUE estimator used by most rloadest models (section 2.2.1 [here](https://pubs.usgs.gov/tm/2005/tm4A5/Pages4_5_6.pdf)). as far as i can remember, all of these methods converge to a simple log() or exp() when uncertainty is 0, but they differ when uncertainty is greater than 0.

therefore bias correction needs to be considered whenever we need to convert the final predictions from linear to log space, at least for the interpolation and composite models. for rloadest models, we might be able to intercept the predictions when they're first in log space rather than going log to linear and back to log.

when we do need a linear-log conversion with uncertainty > 0, and therefore a bias correction, we should assume that the predictions that were being returned by this function 2 weeks ago, which were always in linear space, were also already [sufficiently] bias corrected. the task for this week is to do the bias re-correction, if needed (and i'm actually leaning toward not needed), if and when we take those original linear space predictions and put them back in log space.